README.md

```diff
@@ -190,6 +190,49 @@ Three macros are provided for adding benchmark templates.
 #define BENCHMARK_TEMPLATE2(func, arg1, arg2)
 ```
 
+### Batched processing benchmarks
+
+Sometimes you want to benchmark a batch operation and compare performance for
+different values of batch size. For example, say we have some
+`ParameterizedRoutine` and we want to compare performance as a function of
+the parameter value and not the number of function calls. This could be done
+with:
+
+```c++
+static void BM_ParameterizedRoutine(benchmark::State& state) {
+  const int param = state.range(0);
+  while (state.KeepRunningBatch(param)) {
+    ParameterizedRoutine(param);
+  }
+}
+BENCHMARK(BM_ParameterizedRoutine)->Range(1<<10, 1<<18);
+```
+
+The advantage of using `KeepRunningBatch` is that each batch is counted as
+`param` number of iterations. This means the results for different batch
+sizes are directly comparable as a function of the parameter value.
+
+### Very fast benchmarks
+
+When benchmarking very fast operations (less than 3 ns per iteration), use the
```

Comment:
My reasoning was that the range based for hurts readability a little bit with the unused opaque variable, and the vast majority of cases are not affected by KeepRunning. If that's a nonissue, I can reword the documentation to put the range-based for front and center.

The only other thing is that this library supports C++03 as far as I can tell. So I'm not sure where that leaves things. Add a separate section in the documentation for "old API + C++03"?